src/QuantumGravityBridge.sol

```diff
@@ -198,27 +198,29 @@ contract QuantumGravityBridge is IDAOracle, Initializable, UUPSUpgradeable, Owna
         uint256 _powerThreshold
     ) private pure {
         uint256 cumulativePower = 0;
-
-        for (uint256 i = 0; i < _currentValidators.length; i++) {
-            // If the signature is nil, then it's not present so continue.
-            if (isSigNil(_sigs[i])) {
-                continue;
-            }
-
-            // Check that the current validator has signed off on the hash.
-            if (!verifySig(_currentValidators[i].addr, _digest, _sigs[i])) {
-                revert InvalidSignature();
-            }
-
-            // Sum up cumulative power.
-            cumulativePower += _currentValidators[i].power;
-
-            // Break early to avoid wasting gas.
-            if (cumulativePower >= _powerThreshold) {
-                break;
+        // Note: be cautious when updating the code inside the unchecked block.
+        // Make sure to verify if all the bypassed security checks are respected.
+        unchecked {
+            for (uint256 i = 0; i < _currentValidators.length; i++) {
+                // If the signature is nil, then it's not present so continue.
+                if (isSigNil(_sigs[i])) {
+                    continue;
+                }
+
+                // Check that the current validator has signed off on the hash.
+                if (!verifySig(_currentValidators[i].addr, _digest, _sigs[i])) {
+                    revert InvalidSignature();
+                }
+
+                // Sum up cumulative power.
+                cumulativePower += _currentValidators[i].power;
```

Comment:
The unchecked block also affects this line, along with perhaps the functions above? I'm not sure about the ramifications of disabling overflow protection for this line.

I think the intended way of using `unchecked` with `for` loops is the move the increment _into_ the loop body, at the end.

Reply:
That operation will also never overflow, since even an invalid number can make the bridge halt. That's why I used the `unchecked` for the whole block instead of:
```
unchecked { i++; }
```
at the end.